src/ai/backend/manager/services/groups/types.py

```diff
@@ -1,15 +1,43 @@
 import uuid
 from dataclasses import dataclass, field
 from datetime import datetime
-from typing import Optional, Self
+from typing import Any, Optional, Self, override
 
 from ai.backend.common.types import ResourceSlot, VFolderHostPermissionMap
 from ai.backend.manager.models.group import GroupRow, ProjectType
+from ai.backend.manager.types import Creator, OptionalState, PartialModifier, TriState
+
+
+@dataclass
+class GroupCreator(Creator):
+    name: str
+    domain_name: str
+    type: Optional[ProjectType] = None
+    description: Optional[str] = None
+    is_active: Optional[bool] = None
+    total_resource_slots: Optional[ResourceSlot] = None
+    allowed_vfolder_hosts: Optional[dict[str, str]] = None
+    integration_id: Optional[str] = None
+    resource_policy: Optional[str] = None
+    container_registry: Optional[dict[str, str]] = None
+
+    def fields_to_store(self) -> dict[str, Any]:
+        return {
+            "name": self.name,
+            "domain_name": self.domain_name,
+            "type": self.type,
+            "description": self.description,
+            "is_active": self.is_active,
+            "total_resource_slots": self.total_resource_slots,
+            "allowed_vfolder_hosts": self.allowed_vfolder_hosts,
+            "integration_id": self.integration_id,
+            "resource_policy": self.resource_policy,
+            "container_registry": self.container_registry,
+        }
 
 
 @dataclass
 class GroupData:
-    # TODO: If partial matching test is implemented, we need to remove 'id' from the ignore list
```

Comment:
Is this comment should be deleted?

Reply:
I want to apply a partial test by applying `compare=False`.